src/utils/IdealGraphVisualizer/Coordinator/src/main/java/com/sun/hotspot/igv/coordinator/FolderNode.java

```diff
@@ -56,15 +61,26 @@ public FolderChildren(Folder folder) {
 
         @Override
         protected Node[] createNodes(FolderElement e) {
-             if (e instanceof InputGraph) {
-                return new Node[]{new GraphNode((InputGraph) e)};
+            if (e instanceof InputGraph) {
+                InputGraph g = (InputGraph) e;
+                GraphNode n = new GraphNode(g);
+                graphNode.put(g, n);
+                return new Node[]{n};
             } else if (e instanceof Folder) {
-                 return new Node[]{new FolderNode((Folder) e)};
-             } else {
+                return new Node[]{new FolderNode((Folder) e)};
+            } else {
                 return null;
             }
         }
 
+        @Override
+        protected void destroyNodes(Node[] nodes) {
+            for (Node n : nodes) {
+                // Each node is only present once in the graphNode map.
+                graphNode.values().remove(n);
```

Comment:
Is `destroyNodes()` thread-safe here? graphNode is a HashMap instead of ConcurrentHashMap.